MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -116,11 +116,16 @@ private bool PixelShaderDirty
 
         private readonly ConstantBufferCollection _vertexConstantBuffers = new ConstantBufferCollection(ShaderStage.Vertex, 16);
         private readonly ConstantBufferCollection _pixelConstantBuffers = new ConstantBufferCollection(ShaderStage.Pixel, 16);
+
+        /// <summary>
+        /// Effects can be added to the EffectCache from multiple threads.
+        /// </summary>
+        private readonly object _effectCacheLock = new object();
```

Comment:
You could lock the EffectCache object directly. There is no need for an extra lock object.